crates/uv-resolver/src/requires_python.rs

```diff
@@ -73,24 +74,43 @@ impl RequiresPython {
                 }
             })?;
 
-        // Extract the bounds.
-        let (lower_bound, upper_bound) = range
-            .bounding_range()
-            .map(|(lower_bound, upper_bound)| {
-                (
-                    LowerBound(lower_bound.cloned()),
-                    UpperBound(upper_bound.cloned()),
-                )
-            })
-            .unwrap_or((LowerBound::default(), UpperBound::default()));
-
         // Convert back to PEP 440 specifiers.
         let specifiers = VersionSpecifiers::from_release_only_bounds(range.iter());
 
-        Some(Self {
-            specifiers,
-            range: RequiresPythonRange(lower_bound, upper_bound),
-        })
+        // Extract the bounds.
+        let range = RequiresPythonRange::from_range(&range);
+
+        Some(Self { specifiers, range })
+    }
+
+    /// Split the [`RequiresPython`] at the given version.
+    ///
+    /// For example, if the current requirement is `>=3.10`, and the split point is `3.11`, then
+    /// the result will be `>=3.10 and <3.11` and `>=3.11`.
+    pub fn split(&self, bound: Bound<Version>) -> Option<(Self, Self)> {
+        let RequiresPythonRange(.., upper) = &self.range;
+
+        let left = Range::from_range_bounds((bound, upper.clone().into()));
+        let right = left.complement();
+
+        // Intersect left and right with the existing range.
+        let left = left.intersection(&Range::from(self.range.clone()));
+        let right = right.intersection(&Range::from(self.range.clone()));
+
+        if left.is_empty() || right.is_empty() {
+            None
+        } else {
+            Some((
+                Self {
+                    specifiers: VersionSpecifiers::from_release_only_bounds(left.iter()),
+                    range: RequiresPythonRange::from_range(&left),
+                },
+                Self {
+                    specifiers: VersionSpecifiers::from_release_only_bounds(right.iter()),
+                    range: RequiresPythonRange::from_range(&right),
+                },
+            ))
+        }
```

Comment:
A few unit tests for this routine might be worthwhile. :-)

Reply:
Good call!